### PR TITLE
Download permissions update

### PR DESCRIPTION
**Pull request by jacoblurye**
Revert to the lesser of two evils in terms of IAM conditional bindings limits: 20 per user-trial-datatype combo.

Encode this 20-permission restriction explicitly in `Permissions.insert`.

Also, update the NCI user role's RBAC: they can view all files and upload any manifest without being explicitly granted permission. It's probably worth updating the permission granting UI to convey this fact - NCI biobank users won't need granular permissions anymore.

### CODE REVIEW

File: cidc_api/resources/upload_jobs.py [hunk 1]
```diff
@@ -7,29 +7,17 @@
...
 from cidc_schemas.template_reader import XlTemplateReader
 
 from ..shared import gcloud_client
-from ..shared.auth import (
-    public,
-    requires_auth,
-    get_current_user,
-    authenticate_and_get_user,
-)
+from ..shared.auth import requires_auth, get_current_user, authenticate_and_get_user
```
Comment (jacoblurye) [inline cidc_api/resources/upload_jobs.py:20]: unrelated unused imports I noticed

File: cidc_api/resources/upload_jobs.py [hunk 2]
```diff
@@ -249,13 +236,19 @@ def validate(template, xlsx):
 
 def check_permissions(user, trial_id, template_type):
     """
-    Check that the given user has permissions to access the given trial / template type.
+    Check that the given user has permissions to uplod to the given trial / template type.
```
Comment (jacoblurye) [inline cidc_api/resources/upload_jobs.py:239]:
```suggestion
    Check that the given user has permissions to upload to the given trial / template type.
```

File: tests/resources/test_downloadable_files.py [hunk 3]
```diff
@@ -106,20 +105,24 @@ def test_list_downloadable_files(cidc_api, clean_db, monkeypatch):
...
-
-    # Admin filter queries include any files that fit the criteria
-    res = client.get(f"/downloadable_files?facets=Assay Type|CyTOF|Analysis Results")
-    assert res.status_code == 200
-    assert len(res.json["_items"]) == 1
-    assert res.json["_meta"]["total"] == 1
-    assert res.json["_items"][0]["id"] == file_id_2
+    # Admins and NCI biobank users can view all files regardless of their permissions
+    for role in [CIDCRole.ADMIN.value, CIDCRole.NCI_BIOBANK_USER.value]:
+        print(role)
```
Comment (jacoblurye) [inline tests/resources/test_downloadable_files.py:110]:
```suggestion
```

File: tests/resources/test_upload_jobs.py [hunk 4]
```diff
@@ -610,19 +610,8 @@ def test_upload_manifest(cidc_api, clean_db, monkeypatch, capsys):
 
     client = cidc_api.test_client()
 
-    # Try to upload manifest without permission
+    # NCI users can to upload manifests without explicit permission
```
Comment (jacoblurye) [inline tests/resources/test_upload_jobs.py:613]:
```suggestion
    # NCI users can upload manifests without explicit permission
```

File: tests/resources/test_permissions.py [hunk 5]
```diff
@@ -2,13 +2,16 @@
 from datetime import datetime
 from typing import Tuple
 
+from flask.globals import current_app
+
```
Comment (jacoblurye) [inline tests/resources/test_permissions.py:6]:
```suggestion
```

**Review by curlup — APPROVED** [review]

**Pull request merged**